qmi/instruments/picotech/picoscope3403.py

```diff
@@ -177,46 +215,49 @@ def acquire_by_trigger_and_get_data(
 
             self.set_channel(chan, True, couplings[chan], ranges, 0)
 
-        number_samples = int(time_span / time_base)
+        # Calculate the total number of samples in the time span.
+        number_samples = time_span // sampling_interval
+        _logger.info(f"Timespan = %i Num samples = %i", time_span, number_samples)
+
         if trigger_channel is not None:
             self.stop()
             trig_int = int(self.MAX_SAMPLE_VALUE / max_volts[channels.index(trigger_channel)] * trigger_level)
             self.set_trigger(True, trigger_channel, trig_int, TriggerEdge.RISING)
-            self.run_block(number_samples, number_samples, time_base + 2)
+            self.run_block(number_samples // 2, number_samples - number_samples // 2, time_base)
 
         else:
             self.disable_trigger()
             self.stop()
-            self.run_block(number_samples * 2, 0, time_base + 2)
+            self.run_block(number_samples, 0, time_base)
 
         self.wait_block_ready(timeout=1.5)
         data = self.get_block_data(channels)
-        timetrace = np.arange(0, 2 * number_samples, 1)
+        times = np.arange(0, number_samples, 1) * sampling_interval
         voltages = list()
```

Comment:
this can also be initialised using `[]` instead of `list()`

Reply:
ok